cloud/amazon/route53_health_check.py

```diff
@@ -0,0 +1,352 @@
+        health checks. Required for TCP checks.
+    required: false
+    default: null
+  type:
+    description:
+      - The type of health check that you want to create, which indicates how
+        Amazon Route 53 determines whether an endpoint is healthy.
+    required: true
+    choices: [ 'HTTP', 'HTTPS', 'HTTP_STR_MATCH', 'HTTPS_STR_MATCH', 'TCP' ]
+  resource_path:
+    description:
+      - The path that you want Amazon Route 53 to request when performing
+        health checks. The path can be any value for which your endpoint will
+        return an HTTP status code of 2xx or 3xx when the endpoint is healthy,
+        for example the file /docs/route53-health-check.html.
+
+        * Required for all checks except TCP.
+        * The path must begin with a /
+        * Maximum 255 characters.
+    required: false
+    default: null
+  fqdn:
+    description:
+      - domain name of the endpoint to check
+    required: true
+  string_match:
+    description:
+      - If the check type is HTTP_STR_MATCH or HTTP_STR_MATCH, the string
+        that you want Amazon Route 53 to search for in the response body from
+        the specified resource. If the string appears in the first 5120 bytes
+        of the response body, Amazon Route 53 considers the resource healthy.
+    required: false
+    default: null
+  request_interval:
+    description:
+      - The number of seconds between the time that Amazon Route 53 gets a
+        response from your endpoint and the time that it sends the next
+        health-check request.
+    required: true
+    default: 30
+    choices: [ 10, 30 ]
+  failure_threshold:
+    description:
+      - The number of consecutive health checks that an endpoint must pass or
+        fail for Amazon Route 53 to change the current status of the endpoint
+        from unhealthy to healthy or vice versa.
+    required: true
+    default: 3
+    choices: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
+author: "zimbatm (@zimbatm)"
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Create a health-check for host1.foo.com and use it in record
+- route53_health_check:
+      state: present
+      fqdn: host1.foo.com
+      type: HTTP
+      resource_path: /
+      string_match: "Hello"
+      request_interval: 5
+      failure_threshold: 2
+  record: my_health_check
+
+- route53:
+      action: create
+      zone: "foo.com"
+      type: CNAME
+      record: "www.foo.com"
+      value: host1.foo.com
+      ttl: 30
+      # Routing policy
+      identifier: "host1@www"
+      weight: 100
+      health_check: "{{ my_health_check.health_check.id }}"
+
+# Delete health-check
+- route53_health_check:
+      state: absent
+      fqdn: host1.foo.com
+
+
+'''
+
+import time
+import uuid
+
+try:
+    import boto
+    import boto.ec2
+    from boto import route53
+    from boto.route53 import Route53Connection, exception
+    from boto.route53.healthcheck import HealthCheck
```

Comment:
I am using 2.38.0 at the moment. According to the git logs it should be in 2.27.0+

Should I try to detect and notify accordingly the missing dependency ?